src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_PerfInitPage.js

```diff
@@ -296,24 +313,24 @@ class CJ4_FMC_PerfInitPage {
             { CJ4_FMC_PerfInitPage.ShowPage13(fmc); };
         }
         fmc.onPrevPage = () => {
-            if (fmc.flightPlanManager.getApproachRunway()) {
+            if (fmc.flightPlanManager.getApproachRunway() && fmc.landingQnh > 28 && fmc.landingQnh < 32 && fmc.landingOat && fmc.landingWindDir >= 0 && fmc.landingWindDir <= 360) {
                 CJ4_FMC_PerfInitPage.ShowPage15(fmc);
             }
             else {
                 fmc.showErrorMessage("INVALID");
             }
             };
         fmc.onNextPage = () => {
-            if (fmc.flightPlanManager.getApproachRunway()) {
-                CJ4_FMC_PerfInitPage.ShowPage14(fmc);
+            if (fmc.flightPlanManager.getApproachRunway() && fmc.landingQnh > 28 && fmc.landingQnh < 32 && fmc.landingOat && fmc.landingWindDir >= 0 && fmc.landingWindDir <= 360) {
+                CJ4_FMC_PerfInitPage.ShowPage14(fmc, arrRunwayOutput);
             }
             else {
                 fmc.showErrorMessage("INVALID");
             }
             };
         fmc.updateSideButtonActiveStatus();
     }
-    static ShowPage14(fmc) { //APPROACH REF Page 2
+    static ShowPage14(fmc, arrRunwayOutput = "XX") { //APPROACH REF Page 2
```

Comment:
it is not needed needed to carry over the runway from page 13 to page 14
just get it again in page 14
or extract it to a function

Reply:
Well, it's about 15 lines of code to get the arrRunwayOutput ;-) that's why I didn't want to duplicate it on the next page, but I can... I'm not totally sure on the proper procedure to extract to a function outside the page...a quick lesson and I can probably do that easily.